internal/config/config_type.go

```diff
@@ -42,13 +42,45 @@ var configOptions = []ConfigOption{
 	},
 }
 
-func ConfigOptions() []ConfigOption {
+func ConfigOptions() []configOption {
 	return configOptions
 }
 
-var configValues = map[string][]string{
-	"git_protocol": {"ssh", "https"},
-	"prompt":       {"enabled", "disabled"},
+func ValidateKey(key string) error {
+	for _, configKey := range configOptions {
+		if key == configKey.Key {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("invalid key")
+}
+
+func ValidateValue(key, value string) error {
```

Comment:
Are the exported `Validate*` functions to be used anywhere?

Reply:
They are used in the `config set` command. Specifically in the `setRun` function.